src/lib/plugin.js

```diff
@@ -805,21 +807,46 @@ class FiveBellsLedger extends EventEmitter2 {
   }
 
   * _requestWithCredentials (options) {
-    const authToken = yield this._getAuthToken()
-    return yield request(Object.assign(
-      requestCredentials(this.credentials, authToken),
-      options))
+    // figure out which auth mechanism is supported
+    if (!this.supportedAuth) {
+      this.supportedAuth = yield this._getAuthMechanisms()
+    }
+
+    let requestCreds = null
+    if (this.supportedAuth === 'token') {
+      requestCreds = requestCredentials(this.credentials, yield this._getAuthToken())
+    } else if ((this.supportedAuth === 'basic')) {
+      requestCreds = requestCredentials(this.credentials)
+    }
+
+    return yield request(Object.assign(requestCreds, options))
+  }
+
+  * _getAuthMechanisms () {
+    let tryAuth = function * (options) {
+      let resp = yield request.post(this.ledgerContext.urls.message, options)
+      return +resp.statusCode
+    }.bind(this)
+
+    if ((yield tryAuth({auth: {bearer: 'invalidToken'}})) === 403) {
+      return 'token'
+    } else if ((yield tryAuth({auth: {user: 'abc', pass: 'abc'}})) === 403) {
```

Comment:
Since there are only two options now you can just make this an `else` branch instead of `else if`. Also you shouldn't use fake usernames and passwords like this because then the plugin will behave incorrectly if an account happens to have these values as their credentials.